vembrane/ann_types.py

```diff
@@ -140,12 +141,50 @@ def description(self):
 
 class AnnotationListEntry(AnnotationEntry):
     def __init__(
-        self, name: str, sep: str, typefunc: Callable[[str], Any] = None, **kwargs
+        self,
+        name: str,
+        sep: str,
+        typefunc: Callable[[str], Any] = None,
+        inner_typefunc: Callable[[str], Any] = lambda x: x,
+        **kwargs,
     ):
-        typefunc = typefunc if typefunc else lambda v: [x.strip() for x in v.split(sep)]
+        typefunc = (
+            typefunc
+            if typefunc
+            else lambda v: [inner_typefunc(x.strip()) for x in v.split(sep)]
+        )
         super().__init__(name, typefunc, nafunc=lambda: [], **kwargs)
 
 
+class AnnotationListDictEntry(AnnotationEntry):
+    def __init__(self, name: str, **kwargs):
+        def typefunc(x):
+            key_value_tuples = map(lambda v: v.split(":", maxsplit=1), x.split("&"))
+            mapping = defaultdict(list)
+            for key, value in key_value_tuples:
+                mapping[key].append(value)
+            # make sure non-existent keys do not return list() on __getitem__ calls
+            # in the filter expression (without actually building a new dict from
+            # the defaultdict)
+            mapping.default_factory = None
```

Comment:
So, it is intentional that a user lookup returns `None` for unknown values? (Just asking -- I didn't look at the surrounding code at all. I.e., I also don't know whether we want `None` or `NA` here.)

Reply:
Oh well, good point. Probably something akin to the `nafunc` but with one (unneeded) param? Didn't think that one through